File: java/929/Solution.java
import java.util.Arrays;

class Solution {
    public int numUniqueEmails(String[] emails) {
        return (int) Arrays.stream(emails)
            .map(this::parseEmail)
            .distinct()
            .count();
    }

    private String parseEmail(String email) {
        String[] emailToken = email.split("@");

        String localName = emailToken[0];
        String domainName = emailToken[1];

        String forwardedLocalName = parseLocalName(localName);

        return forwardedLocalName + "@" + domainName;
    }

    private String parseLocalName(String localName) {
        return localName.replaceAll("\\.", "")
            .replaceAll("\\+.*", "");
    }
}

File: python3/929/Solution.py
import re

class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        return len(set(map(self.parseEmail, emails)))

    def parseEmail(self, email: str) -> str:
        localName, domainName = email.split("@")

        forwardedLocalName = self.parseLocalName(localName)

        return f"{forwardedLocalName}@{domainName}"

    def parseLocalName(self, localName: str) -> str:
        return re.sub(r"\.", "", re.sub(r"\+.*", "", localName))
